### PR TITLE
Use EPP instead of ERB for some templates

### PR DESCRIPTION
**Pull request by cocker-cc**
As a Preparation for making this Module "Sensitive"-aware, we prefer to
use EPP instead of ERB in such Situations, where "Sensitive" could come
into Play, as EPP is able to handle Data of Type Sensitive natively.

### CODE REVIEW

File: templates/hammer_root.yml.epp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<%- |
+  Optional[String] $username,
+  Optional[String] $password,
+| -%>
+:foreman:
+  # Credentials. You'll be asked for the interactively if you leave them blank here
+  :username: '<%= if $username { $username } else { 'undef'} %>'
+  :password: '<%= if $password { $password } else { 'undef'} %>'
```
Comment (ekohl) [inline templates/hammer_root.yml.epp:8]: I'm not sure if `undef` is really intended. You're probably right that that was the previous behavior but it wasn't really intended. IIRC Puppet < 4 evaluated it to an empty string and we didn't have a test for it. I doubt hammer handles `undef`.

tl;dr: I think it should be this:

```suggestion
  :username: '<%=$username %>'
  :password: '<%= $password %>'
```

File: manifests/plugin/supervisory_authority.pp [hunk 2]
```diff
@@ -42,6 +42,21 @@
...
+        log_level             => $log_level,
+        pool_size             => $pool_size,
+        api_buffer_size       => $api_buffer_size,
+        api_request_size      => $api_request_size,
+        api_request_time      => $api_request_time,
+        transaction_max_spans => $transaction_max_spans,
+        http_compression      => $http_compression,
+        metrics_interval      => $metrics_interval,
+      }
+    ),
```
Comment (ekohl) [inline manifests/plugin/supervisory_authority.pp:60]: Untested, but I think it can actually not use any template and use this instead:
```suggestion
    config => foreman::to_symbolized_yaml(
      {
        server_url            => $server_url,
        secret_token          => $secret_token,
        service_name          => $service_name,
        log_level             => $log_level,
        pool_size             => $pool_size,
        api_buffer_size       => $api_buffer_size,
        api_request_size      => $api_request_size,
        api_request_time      => $api_request_time,
        transaction_max_spans => $transaction_max_spans,
        http_compression      => $http_compression,
        metrics_interval      => $metrics_interval,
      }
    ),
```

**Review by ekohl — COMMENTED** [review]
Some inline suggestions for improvements. While this doesn't convert every file, I'm actually content with this as a start.

File: templates/hammer_root.yml.epp (hunk 1 above)
Comment (cocker-cc) [reply templates/hammer_root.yml.epp]: Suggestion implemented.

File: manifests/plugin/supervisory_authority.pp (hunk 2 above)
Comment (cocker-cc) [reply manifests/plugin/supervisory_authority.pp:60]: Maybe You can test it later, if the Function can handle `Sensitive[String] $secret_token`.

Comment (ekohl) [reply manifests/plugin/supervisory_authority.pp:60]: You don't need to because you can unwrap when passing the hash.

Comment (cocker-cc) [reply manifests/plugin/supervisory_authority.pp:60]: I thought, this was the whole Point of the Story: Not having to spread random `unwrap`s in the Code.  If the Function `foreman::to_symbolized_yaml` cannot handle `Sensitive`, then it has to be empowered to do so, and just like the Function `epp` it should return `Sensitive` then.

Comment (cocker-cc) [reply manifests/plugin/supervisory_authority.pp:60]: Besides that, `foreman::to_symbolized_yaml` generates a symbolized(!) YAML like… 
```
---
:a: A
:b:
  x: X
  y: Y
:c:
- 1
- 2
- 3

```
…  (note the leading `:`)  I guess that's not, what the Configfile wants.

Comment (ekohl) [reply manifests/plugin/supervisory_authority.pp:60]: Have a look at the template: it's exactly what it wants.

Comment (cocker-cc) [reply manifests/plugin/supervisory_authority.pp:60]: Yes, You are right 👍   

[I upgraded Your Function](https://github.com/cocker-cc/puppet-foreman/compare/6c902a4f1f484137ebccafe4d3390881fd0b3d61..1fba024a8d226a0d7c77822c09bce363bc107f96) to be sensitive-aware.  
What do You think of a Pullrequest?

Comment (ekohl) [reply manifests/plugin/supervisory_authority.pp:60]: For our use case right now I think it only complicates matters. I'd prefer to unwrap it in Puppet code.

**Comment by ekohl** [comment]
I decided to push https://github.com/theforeman/puppet-foreman/pull/964 which is what I meant. I also left a comment where I think we should unwrap it.

File: manifests/plugin/supervisory_authority.pp (hunk 2 above)
Comment (cocker-cc) [reply manifests/plugin/supervisory_authority.pp:60]: If You want to, save my Code somewhere.  I will delete it, after this Story's End.

**Comment by ekohl** [comment]
I've merged #964 so would you mind rebasing to resolve the conflict?

**Comment by cocker-cc** [comment]
> I've merged #964 so would you mind rebasing to resolve the conflict?

Done.

**Review by ekohl — APPROVED** [review]
I'll let the tests complete, but otherwise :+1:

**Comment by ekohl** [comment]
And those tests are green. Thanks!
